src/pixie/fileformats/qoi.nim

```diff
@@ -15,158 +14,150 @@ const
   opRun = 0b11000000'u8
 
 type
-  Colorspace* = enum sRBG = 0, linear = 1
+  Colorspace* = enum
+    sRBG = 0
+    Linear = 1
+
   Qoi* = ref object
     ## Raw QOI image data.
-    data*: seq[ColorRGBA]
     width*, height*, channels*: int
     colorspace*: Colorspace
+    data*: seq[ColorRGBA]
 
   Index = array[indexLen, ColorRGBA]
 
 func hash(p: ColorRGBA): int =
   (p.r.int * 3 + p.g.int * 5 + p.b.int * 7 + p.a.int * 11) mod indexLen
 
-func toImage*(qoi: Qoi): Image =
+func newImage*(qoi: Qoi): Image =
   ## Converts raw QOI data to `Image`.
   result = newImage(qoi.width, qoi.height)
   copyMem(result.data[0].addr, qoi.data[0].addr, qoi.data.len * 4)
   result.data.toPremultipliedAlpha()
 
-func toQoi*(img: Image; channels: range[3..4]): Qoi =
```

Comment:
be consistent with png's raw api, do not take in channels without tests (this is not run in any tests)